asserts/snap_asserts.go

```diff
@@ -121,46 +121,51 @@ func assembleSnapDeclaration(assert assertionBase) (Assertion, error) {
 		return nil, err
 	}
 
-	refControl, err := checkStringList(assert.headers, "refresh-control")
+	timestamp, err := checkRFC3339Date(assert.headers, "timestamp")
 	if err != nil {
 		return nil, err
 	}
 
+	var refControl []string
 	var plugRules map[string]*PlugRule
-	plugs, err := checkMap(assert.headers, "plugs")
-	if err != nil {
-		return nil, err
-	}
-	if plugs != nil {
-		plugRules = make(map[string]*PlugRule, len(plugs))
-		for iface, rule := range plugs {
-			plugRule, err := compilePlugRule(iface, rule)
-			if err != nil {
-				return nil, err
-			}
-			plugRules[iface] = plugRule
+	var slotRules map[string]*SlotRule
+
+	// guard evolvable complex parts
+	if assert.SupportedFormat() {
```

Comment:
Feels like a slightly fragile approach. If we don't understand the format, we should probably assume we also don't understand why we don't understand the format. Perhaps we might control this on the general part of the parsing by using the format header itself to guard the information that changed. Or perhaps prevent the assertion from being read altogether?